akka-remote/src/main/scala/akka/remote/artery/ArteryTransport.scala

```diff
@@ -352,3 +359,19 @@ private[remote] class ArteryTransport(_system: ExtendedActorSystem, _provider: R
 
 }
 
+object ArteryTransport {
+
+  /**
+   * Internal API
+   * @param hostname
+   * @return A port that is hopefully available
+   */
+  private[artery] def autoSelectPort(hostname: String): Int = {
```

Comment:
we are mixing a bit with the visibility, I think we should decide on `private[remote]` as the default if no other reason
I'm to blame for several `private[akka]`